File: backend/app/schemas/diligence.py
from datetime import datetime
from typing import List, Literal, Optional, Dict, Any

from pydantic import BaseModel, Field


class Metric(BaseModel):
    name: str
    value: str | float | int
    trend: Literal["up", "flat", "down"] = "flat"
    confidence: float = Field(ge=0, le=1, default=0.7)


class RiskFlag(BaseModel):
    code: str
    severity: Literal["low", "medium", "high"]
    description: str


class ExternalDataSource(BaseModel):
    """Summary of data from an external source."""
    source: str
    status: str
    data: Dict[str, Any] = Field(default_factory=dict)


class DiligenceSummary(BaseModel):
    profile_id: str
    score: float = Field(ge=0, le=100)
    metrics: List[Metric] = Field(default_factory=list)
    risks: List[RiskFlag] = Field(default_factory=list)
    narrative: str | None = None
    strengths: List[str] = Field(default_factory=list, description="LLM-generated strengths / what's good about the company")
    concerns: List[str] = Field(default_factory=list, description="LLM-generated concerns / what's bad or risky")
    sources_used: List[str] = Field(default_factory=list)
    external_data: Optional[Dict[str, Any]] = None
    generated_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/src/lib/api/diligence.ts
/**
 * Due Diligence API endpoints
 */

import apiClient, { getErrorMessage } from '../api-client';

export interface Metric {
  name: string;
  value: string | number;
  trend: 'up' | 'flat' | 'down';
  confidence: number;
}

export interface RiskFlag {
  code: string;
  severity: 'low' | 'medium' | 'high';
  description: string;
}

export interface ExternalData {
  apollo?: {
    company_name?: string;
    industry?: string;
    employee_count?: number;
    total_funding?: number;
    latest_funding_round?: string;
    location?: string;
    technologies?: string[];
    linkedin_url?: string;
  };
  email_verification?: {
    result?: string;
    score?: number;
    disposable?: boolean;
    webmail?: boolean;
  };
  ai_analysis?: {
    summary?: string;
    business_model?: string;
    strengths?: string[];
    risks?: string[];
    competitors?: string[];
    market_size?: string;
  };
  pdl?: {
    company_name?: string;
    industry?: string;
    employee_count?: number;
    employee_range?: string;
    total_funding?: number;
    latest_funding_round?: string;
    company_type?: string;
    location?: string;
    linkedin_url?: string;
    tags?: string[];
  };
  founder?: {
    full_name?: string;
    headline?: string;
    company?: string;
    linkedin_url?: string;
    skills?: string[];
    experience?: Array<{
      title?: string;
      company?: string;
      start_date?: string;
      end_date?: string;
    }>;
    education?: Array<{
      school?: string;
      degree?: string;
      field?: string;
    }>;
  };
}

export interface DiligenceSummary {
  profile_id: string;
  score: number;
  metrics: Metric[];
  risks: RiskFlag[];
  narrative: string | null;
  strengths?: string[];
  concerns?: string[];
  sources_used: string[];
  external_data: ExternalData | null;
  generated_at: string;
}

export const diligenceApi = {
  /**
   * Get due diligence summary for a profile
   */
  async getSummary(profileId: string, forceRefresh = false): Promise<DiligenceSummary> {
    try {
      const response = await apiClient.get<DiligenceSummary>(`/diligence/${profileId}`, {
        params: { force_refresh: forceRefresh },
      });
      return response.data;
    } catch (error) {
      throw new Error(getErrorMessage(error));
    }
  },
};
